### PR TITLE
Fixes bugs, adds new functions, moves  dtypes to `jumpy.numpy` package

### LINKED ISSUE #29: Non-deterministic behavior jit vs non-jit
I get different results from the `jp.random_split` function after the function is jit-ted. 

```python
import jumpy as jp
import jax
import numpy as onp


def jp_prng(rng):
	return jp.random_split(rng)   # jit --> uses jax, non-jit --> uses numpy.


def jax_prng(rng):
	return jax.random.split(rng)


jp_prng_jit = jax.jit(jp_prng)
jax_prng_jit = jax.jit(jax_prng)

seed = jp.random_prngkey(0)  # always returns a numpy array (regardless of input type).

jax_key_jit = jax_prng_jit(seed)
jax_key = jax_prng(seed)

jp_key_jit = jp_prng_jit(seed)
jp_key = jp_prng(seed)

print(f"jax: {onp.isclose(jax_key_jit, jax_key).all()}")   # --> jax: True
print(f"jumpy: {onp.isclose(jp_key, jp_key_jit).all()}")  # --> jumpy: False


```
This behavior is caused by `jp.random_prngkey(0)` which always returns a numpy array. Therefore, in the non-jit version `jp_prng` uses its numpy implementation, while in jit it uses the jax one (which results in different keys).

A possible solution could be to have the random_prngkey select the backend based on the seed type (int, jax.int32, np.int32). This would then lead to deterministic behavior for jit/non-jit when a key is provided with a jax type.

The random_prngkey function would need to be changed as follows:

```python

def random_prngkey(seed: jp.int32) -> jp.ndarray:
    """Returns a PRNG key given a seed."""
    if jp._which_np(seed) is jnp:   # NOTE: selects backend based on seed type.
        return jax.random.PRNGKey(seed)
    else:
        rng = onp.random.default_rng(seed)
        return rng.integers(low=0, high=2**32, dtype="uint32", size=2)

```

### PR DESCRIPTION
**Pull request by bheijden**
This PR addresses the following:

- bug fixes for `jumpy.lax.fori_loop` and `jumpy.lax.cond`. 
- Moves all dtypes from `jumpy` to `jumpy.numpy`, following the module structure of `jax.numpy` and `numpy`.
- Added some new implementations to `jumpy.lax module` (e.g.`stop_gradient` and `switch`).
- Fixes non-deterministic behavior with jitted vs non-jitted functions. Closes #29.

### CODE REVIEW

File: jumpy/__init__.py [hunk 1]
```diff
@@ -1,39 +1,22 @@
...
     # === primitives ===
-    "dtype",
-    "pi",
-    "inf",
-    "float32",
-    "int32",
-    "ndarray",
+    "jparray",
     # === sub-modules ===
+    "core",
```
Comment (pseudo-rnd-thoughts) [inline jumpy/__init__.py:19]: We don't need to include `core` if we don't want users to do `jumpy.core.*`. It should be still possible for users to do `from jumpy.core import X`

File: jumpy/lax.py [hunk 2]
```diff
@@ -2,21 +2,32 @@
...
 except ImportError:
     jax, jnp = None, None
 
 
-__all__ = ["cond", "fori_loop", "scan", "top_k", "while_loop"]
+__all__ = [
+    "cond",
+    "switch",
+    "fori_loop",
+    "scan",
```
Comment (pseudo-rnd-thoughts) [inline jumpy/lax.py:26]: Could the functions be in alphabetical order?

**Review by pseudo-rnd-thoughts — CHANGES_REQUESTED** [review]
Other than the indexing code, I think this is good.
Could you remove the indexing code then we can merge it

File: jumpy/__init__.py (hunk 1 above)
Comment (bheijden) [reply jumpy/__init__.py:19]: I actually like the `jumpy.core.*` syntax, but if your against I'm fine with removing it!

File: jumpy/lax.py (hunk 2 above)
Comment (bheijden) [reply jumpy/lax.py:26]: yes, will do!

File: jumpy/__init__.py (hunk 1 above)
Comment (pseudo-rnd-thoughts) [reply jumpy/__init__.py:17]: I just realised a major issue, `jax.core.*` already exists, therefore, I think we need to rename the file. 
Also if you want to keep `jumpy.core.*` Im happy to do keep it

Comment (bheijden) [reply jumpy/__init__.py:17]: I find it hard to find a proper alternative. All the logical names seem to already be in use by jax (e.g. `util`). 

Maybe it is not so bad if we keep it in `jumpy.core`. As long as the methods we add to the module either have a unique name (w.r.t `jax.core`), or the methods should have the same signature and have the same function?

Comment (pseudo-rnd-thoughts) [reply jumpy/__init__.py:17]: Ok, we can leave it as it is. Maybe it makes more sense to use `utils` rather than `core`

**Review by pseudo-rnd-thoughts — APPROVED** [review]
